index.d.ts

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+
+type DateInfo = {
+    day: number;
+    month: number;
+    year: number;
+}
+
+type Range = {
+    from: DateInfo | null;
+    To: DateInfo | null;
+}
+
+export interface DatePickerProps {
+    value: Range | DateInfo[] | null;

```

Comment:
As in the react type definition the [default value for the `useState` function is `undefined`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react/index.d.ts#L865) let's use `undefined` instead of `null` in the whole definition.